zebra-network/src/peer_set/set/tests.rs

```diff
@@ -21,10 +18,7 @@ use zebra_chain::{
 
 use super::MorePeers;
 use crate::{
-    peer::{
-        CancelHeartbeatTask, Client, ClientRequest, ErrorSlot, LoadTrackedClient,
-        MinimumPeerVersion,
-    },
+    peer::{LoadTrackedClient, MinimumPeerVersion, MockedClientHandle},
```

Comment:
Question: How do we know when we can get accurate results from testing `PeerSet<MockedClientHandle>`? How do we know when we need to test `PeerSet<peer::Client>` for good test results?

Where are the tests that make sure that `MockedClientHandle` behaves like `peer::Client`? How much is it ok for the implementations to diverge?

Can you add some documentation explaining when to use each of them?

Reply:
It should be exactly the same. `MockedClientHandle::new()` returns `(Self, Client)`, and the `Client` is what is passed to the `PeerSet`. The handle is used just to access things inside the `Client`.